src/main/java/listeners/MainEventListener.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    private void maketopic(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Topic topic = new Topic(args[0]);
+        server.createTopic(topic);
+
+        channel.sendMessage(String.format(
+                "%s Topic role \"%s\" has been created.",
+                member.getAsMention(),
+                args[0])).complete();
+    }
+
+    private void deletetopic(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Arrays.stream(server.getTopics())                   // loop over all topics...
+                .filter(t -> t.getName().equals(args[0]))   // ...if this topic's name is args[0]...
+                .findFirst()
+                .ifPresent(server::deleteTopic);            // ...delete it
+
+        channel.sendMessage(String.format(
+                "%s Topic role \"%s\" has been deleted.",
+                member.getAsMention(),
+                args[0])).complete();
+    }
+
+    private void showtopics(Member member, TextChannel channel, Server server, String[] args) {
+        String topicList = Arrays.stream(server.getTopics())
+                .map(Topic::getName)
+                .collect(Collectors.joining("\n"));
+
+        channel.sendMessage(String.format(
+                "%s List of topics:\n%s",
+                member.getAsMention(),
+                topicList)).complete();
+    }
+
+    private void queue(Member member, TextChannel channel, Server server, String[] args) {
+        Optional<Topic> topicOptional = checkTopicExists(member, channel, server, args[0]);
+        if (topicOptional.isEmpty()) return;
+
+        Topic topic = topicOptional.get();
+        if (topic.isInQueue(member)) {
+            topic.removeFromQueue(member);
+            channel.sendMessage(String.format(
+                    "%s has left the \"%s\" queue.",
+                    member.getAsMention(),
+                    args[0])).complete();
+        } else {
+            topic.addToQueue(member);
+            channel.sendMessage(String.format(
+                    "%s has joined the \"%s\" queue.",
+                    member.getAsMention(),
+                    args[0])).complete();
+        }
+    }
+
+    private void ready(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Optional<Topic> topicOptional = checkTopicExists(member, channel, server, args[0]);
+        if (topicOptional.isEmpty()) return;
+
+        Topic topic = topicOptional.get();
+        Member mentee = topic.getNextFromQueue();
+        channel.sendMessage(String.format(
+                "%s is ready for %s.",
+                member.getAsMention(),
+                mentee.getAsMention())).complete();
+    }
+
+    private void showqueue(Member member, TextChannel channel, Server server, String[] args) {
```

Comment:
Maybe make it so this command can only be called from a specific channel. It feels a little undesirable to just post the names of the people waiting in a chat anyone can access.

Reply:
You make a good point, but this falls outside of the scope of this PR. I will create a new issue for it after this PR gets merged.

Reply:
Dope, just popped into my head at that moment

Reply:
We can def re-evaluate the exact functionality sometime this week and at next week's meeting!